sql/core/src/main/scala/org/apache/spark/sql/execution/UnsafeShuffledRowRDD.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution
+
+import java.io.InputStream
+
+import scala.util.control.NonFatal
+
+import org.apache.spark.shuffle.BinaryShuffleWriter
+import org.apache.spark.shuffle.unsafe.UnsafeShuffleWriter
+import org.apache.spark.sql.catalyst.expressions.UnsafeRow
+import org.apache.spark.{InterruptibleIterator, SparkEnv, TaskContext}
+import org.apache.spark.io.CompressionCodec
+import org.apache.spark.util.CompletionIterator
+import org.apache.spark.rdd.{BinaryShuffledRDD, RDD}
+
+
+class UnsafeShuffledRowRDD(
```

Comment:
Here is an example of code in Spark SQL which implements a binary shuffle. Note that the specifics of this class are not particularly interesting, since it currently just calls into other shuffle components, but the idea here was to illustrate how the write-path of the shuffle can be customized without creating a new ShuffleManager.